crates/red_knot_python_semantic/src/types.rs

```diff
@@ -435,6 +436,14 @@ pub enum Type<'db> {
     // TODO protocols, callable types, overloads, generics, type vars
 }
 
+impl TryFrom<usize> for Type<'_> {
+    type Error = TryFromIntError;
+
+    fn try_from(value: usize) -> Result<Self, Self::Error> {
+        i64::try_from(value).map(Type::IntLiteral)
+    }
+}
```

Comment:
I'm not a massive fan of this. Converting "from a usize to a Type" doesn't feel like the right way of thinking about what we're doing when we attempt to create a `Type::IntLiteral` from a numeric value that has type `usize`. It also doesn't save us many lines; I think I'd prefer to just write it out:

```diff
--- a/crates/red_knot_python_semantic/src/types.rs
+++ b/crates/red_knot_python_semantic/src/types.rs
@@ -1,5 +1,4 @@
 use std::hash::Hash;
-use std::num::TryFromIntError;
 
 use indexmap::IndexSet;
 use itertools::Itertools;
@@ -436,14 +435,6 @@ pub enum Type<'db> {
     // TODO protocols, callable types, overloads, generics, type vars
 }
 
-impl TryFrom<usize> for Type<'_> {
-    type Error = TryFromIntError;
-
-    fn try_from(value: usize) -> Result<Self, Self::Error> {
-        i64::try_from(value).map(Type::IntLiteral)
-    }
-}
-
 impl<'db> Type<'db> {
     pub const fn is_never(&self) -> bool {
         matches!(self, Type::Never)
@@ -1371,11 +1362,15 @@ impl<'db> Type<'db> {
     /// This is used to determine the value that would be returned
     /// when `len(x)` is called on an object `x`.
     fn len(&self, db: &'db dyn Db) -> Option<Type<'db>> {
-        match self {
-            Type::BytesLiteral(bytes) => return bytes.len(db).try_into().ok(),
-            Type::StringLiteral(string) => return string.python_len(db).try_into().ok(),
-            Type::Tuple(tuple) => return tuple.elements(db).len().try_into().ok(),
-            _ => {}
+        let usize_len = match self {
+            Type::BytesLiteral(bytes) => Some(bytes.len(db)),
+            Type::StringLiteral(string) => Some(string.python_len(db)),
+            Type::Tuple(tuple) => Some(tuple.elements(db).len()),
+            _ => None,
+        };
+
+        if let Some(usize_len) = usize_len {
+            return usize_len.try_into().ok().map(Type::IntLiteral);
         }
```